src/components/button.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { MouseEvent } from "react";
+import "./button.css";
+
+// HINT: Use this in ButtonProps to determine how the button is styled
+export enum ButtonType {
+  "button-green",
+  "button-white",
+  "button-clear",
+  "button-clearAlt",
+}
+
+// HINT: Use this in ButtonProps to determine how the button text is styled
+export enum ButtonTextType {
+  "button-text-white",
+  "button-text-green",
+  "button-text-grey",
+}
+
+interface ButtonProps {
+  text: string;
+  buttonType: string;
```

Comment:
This might be a potential application for the enum exports. Like the hint mentions, you want to use the enums here so that you can declare the type for each prop.

There are a few changes necessary to make this work:
1. change this line to `buttonType: ButtonType`
2. when you use the button component in `App.tsx`, use the enum instead of the string
```
// keep in mind you'd have to import ButtonType and ButtonTextType (which is why you export the enums!)
<Button
          text='CONTINUE'
          buttonType={ButtonType['button-green']}
          textType={ButtonTextType['button-text-white']}
          onPress={() => console.log("I'm the second.")}
     />
```

This way, the ButtonProps will ensure that you only pass in a buttonType of one of the possible options from the enum, instead of any random string. (same applies to ButtonTextType)

Reply:
One thing I'll also note (which is also why this might've been confusing), is that the way the enum is structured in the boilerplate code might not be ideal. Here's another way that might make more sense:

```
export enum ButtonType {
  PRIMARY = "button-green", 
  WHITE = "button-white",
  CLEAR = "button-clear",
  CLEAR_ALT = "button-clearAlt",
}
```

_Side note: naming the first color GREEN instead of PRIMARY would also be fine, but it's pretty common for people to name color styles fo the context of their usage (e.g. primary, secondary). See [Material Design on Color](https://material.io/design/color/the-color-system.html#color-usage-and-palettes) for more on this. If you name it by its function, we could change the app-wide theme to be based on say blue instead of green, but we don't have to manually change every .GREEN to .BLUE._

Then when you use the button component, you could just do this:
```
<Button
          text='CONTINUE'
          buttonType={ButtonType.PRIMARY}
          textType={ButtonTextType.WHITE}
          onPress={() => console.log("I'm the second.")}
     />

```
which is probably a lot more intuitive (you don't ever actually type the whole id string `button-text-white`.

Hopefully this makes sense and didn't just confuse you more :'') let me know if you have any qs!